pkg/apis/servicecatalog/validation/broker.go

```diff
@@ -218,6 +221,29 @@ func validateCommonServiceBrokerSpec(spec *sc.CommonServiceBrokerSpec, fldPath *
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(scfeatures.CatalogRestrictions) {
+		// TODO: could validate if the fields being selected are on the approve list, but this will require breaking
+		// apart the label selector.
+		if spec.CatalogRestrictions != nil && len(spec.CatalogRestrictions.ServiceClass) > 0 {
+			// confirm that the restrictions can turn into a predicate.
+			_, err := filter.CreatePredicate(spec.CatalogRestrictions.ServiceClass)
+			if err != nil {
+				commonErrs = append(commonErrs,
+					field.Invalid(fldPath.Child("catalogRestrictions", "serviceClass"),
+						spec.CatalogRestrictions.ServiceClass, err.Error()))
+			}
+		}
+		if spec.CatalogRestrictions != nil && len(spec.CatalogRestrictions.ServicePlan) > 0 {
+			// confirm that the restrictions can turn into a predicate.
+			_, err := filter.CreatePredicate(spec.CatalogRestrictions.ServicePlan)
+			if err != nil {
+				commonErrs = append(commonErrs,
+					field.Invalid(fldPath.Child("catalogRestrictions", "servicePlan"),
+						spec.CatalogRestrictions.ServiceClass, err.Error()))
+			}
+		}
+	}
```

Comment:
One thing I just thought of: I think we should not allow you to set these fields if the feature gate isn't enabled.

Reply:
meh, I am not sure about that. I could see a use for setting this up then turning on or off the feature.

Reply:
It's conventional not to allow fields for a feature that isn't enabled.

Reply:
If you accept the fields, users will expect them to work.